tensorboard/plugins/image/summary_v2.py

```diff
@@ -27,6 +27,29 @@
 def image(name, data, step=None, max_outputs=3, description=None):
     """Write an image summary.
 
+    This method writes a set of images to the current log directory.
+    The written data appears in TensorBoard's 'Images' dashboard.
+
+    This example writes 2 random grayscale images:
+
+    ```
+    writer = tf.summary.create_file_writer('logs')
+    with writer.as_default():
+      image1 = tf.random.normal(shape=[8, 8])
+      image2 = tf.random.normal(shape=[8, 8])
+      tf.summary.image("grayscale noise", [image1, image2], step=0)
+    ```
+
+    When image data is outside of the range [0, 1), please rescale the data
+    before writing the summary, to avoid a clipped result.
+
+    ```
+    rgb_image = tf.constant([
```

Comment:
This image violates the above constraint.  Specifically, the constraints specifies the range [0, 1) with the open value at the top of the range.  However, 255/255 is exactly 1.   Did you mean `[0, 1]` (note square bracket).

Also, be sure those values are floats and not ints, otherwise you will end up with an image which has values exactly 0 or 1.

Reply:
Good catch, thanks.  I did some testing and found that floats are clipped, uint8's are preserved (already in range [0,255]), and things are strange when converting from a signed type.

Internally we use https://www.tensorflow.org/api_docs/python/tf/image/convert_image_dtype, whose docs say that it expects floating point values to have range [0,1), with an exclusive end of the range.  However, testing out the summary API, float32 and float64 values of `1.0` are indeed accepted (treated like rgb 255 in range [0,255]), so I've updated the docs to say `[0,1]` inclusive.

Reply:
FWIW, I think the fact that it accepts `1.0` is an artifact of the saturation / wrap-around.  So I don't think the docs for `convert_image_dtype` are incorrect; semantically the range probably is supposed to be `[0, 1)`, but `1.0` still works because it gets clipped, as does any float that's `>= 1.0`.